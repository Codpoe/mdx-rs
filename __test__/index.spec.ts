import { readFileSync } from "fs";
import { describe, test, expect } from "vitest";
import path from "path";

import { compile } from "../index.js";

describe("compile", () => {
  test("container grammer", async (t) => {
    let result = await compile(
      readFileSync(path.join(__dirname, "./container.md"), "utf8"),
      "xxx.mdx"
    );

    expect(result).toMatchSnapshot();
  });
});
