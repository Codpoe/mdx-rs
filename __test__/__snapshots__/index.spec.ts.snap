// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile > should render container content correctly 1`] = `
"<div>
  <div>TIP</div>
  <div>
    <p>First line <code>block</code></p>
    <p>Second line <code>block2</code></p>
  </div>
</div>
<div>
  <div>TIP</div>
  <div>
    <ul>
      <li>List item 1 <code>block</code>。</li>
      <li>List item 2 <code>block2</code>。</li>
    </ul>
  </div>
</div>
<div>
  <div>TIP</div>
  <div>
    <p>Some code:</p>
    <pre><code>console.log('hello world');
</code></pre>
  </div>
</div>
"
`;

exports[`compile > should render container content correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\",
        ul: \\"ul\\",
        li: \\"li\\",
        pre: \\"pre\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"First line \\"}<_components.code>{\\"block\\"}</_components.code></_components.p>{\\"\\\\n\\"}<_components.p>{\\"Second line \\"}<_components.code>{\\"block2\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"modern-directive-content\\">{\\"\\\\n\\"}<_components.ul>{\\"\\\\n\\"}<_components.li>{\\"List item 1 \\"}<_components.code>{\\"block\\"}</_components.code>{\\"。\\"}</_components.li>{\\"\\\\n\\"}<_components.li>{\\"List item 2 \\"}<_components.code>{\\"block2\\"}</_components.code>{\\"。\\"}</_components.li>{\\"\\\\n\\"}</_components.ul>{\\"\\\\n\\"}</_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"modern-directive-content\\">{\\"\\\\n\\"}<_components.p>{\\"Some code:\\"}</_components.p>{\\"\\\\n\\"}<_components.pre><_components.code className=\\"language-ts\\">{\\"console.log('hello world');\\\\n\\"}</_components.code></_components.pre>{\\"\\\\n\\"}</_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container title correctly 1`] = `
"<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container title correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"\\\\nThis is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"\\\\nThis is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container type correctly 1`] = `
"<div>
  <div>TIP</div>
  <div>
    <p>This is a <code>block</code> of type <code>tip</code></p>
  </div>
</div>
<div>
  <div>INFO</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>NOTE</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>WARNING</div>
  <div>
    <p>This is a <code>block</code> of type <code>warning</code></p>
  </div>
</div>
<div>
  <div>DANGER</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
<div>
  <div>CAUTION</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container type correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"tip\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive info\\"><_components.div className=\\"modern-directive-title\\">{\\"INFO\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive note\\"><_components.div className=\\"modern-directive-title\\">{\\"NOTE\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive warning\\"><_components.div className=\\"modern-directive-title\\">{\\"WARNING\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"warning\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive danger\\"><_components.div className=\\"modern-directive-title\\">{\\"DANGER\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive caution\\"><_components.div className=\\"modern-directive-title\\">{\\"CAUTION\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container type with space correctly 1`] = `
"<div>
  <div>TIP</div>
  <div>
    <p>This is a <code>block</code> of type <code>tip</code></p>
  </div>
</div>
<div>
  <div>TIP</div>
  <div>
    <p>This is a <code>block</code> of type <code>tip</code></p>
  </div>
</div>
<div>
  <div>INFO</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>NOTE</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>WARNING</div>
  <div>
    <p>This is a <code>block</code> of type <code>warning</code></p>
  </div>
</div>
<div>
  <div>DANGER</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
<div>
  <div>CAUTION</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container type with space correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"tip\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive tip\\"><_components.div className=\\"modern-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"tip\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive info\\"><_components.div className=\\"modern-directive-title\\">{\\"INFO\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive note\\"><_components.div className=\\"modern-directive-title\\">{\\"NOTE\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive warning\\"><_components.div className=\\"modern-directive-title\\">{\\"WARNING\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"warning\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive danger\\"><_components.div className=\\"modern-directive-title\\">{\\"DANGER\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"modern-directive caution\\"><_components.div className=\\"modern-directive-title\\">{\\"CAUTION\\"}</_components.div><_components.div className=\\"modern-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;
